wallet-core/src/ffi/mem.rs

```diff
@@ -40,17 +40,20 @@ pub fn free(ptr: u32, len: u32) {
     }
 }
 
-/// Checks and deserializes a value from the given po
-pub unsafe fn from_buffer<T>(ptr: *const u8) -> Result<T, ErrorCode>
+/// Read a buffer from the given pointer.
+pub unsafe fn read_buffer<'a>(ptr: *const u8) -> &'a [u8] {
+    let len = slice::from_raw_parts(ptr, 4);
+    let len = u32::from_le_bytes(len.try_into().unwrap()) as usize;
```

Comment:
Do we have a panic handler somewhere? I dont see it, ideally we shouldn't ever panic and return something from the ErrorCode